src/main/java/org/cryptomator/ui/health/CheckExecutor.java

```diff
@@ -94,18 +95,21 @@ protected void running() {
 		@Override
 		protected void cancelled() {
 			c.setState(Check.CheckState.CANCELLED);
+			tasksToExecute.remove(this);
 		}
 
 		@Override
 		protected void succeeded() {
 			c.setState(Check.CheckState.SUCCEEDED);
 			c.setHighestResultSeverity(highestResultSeverity);
+			tasksToExecute.remove(this);
 		}
 
 		@Override
 		protected void failed() {
 			c.setState(Check.CheckState.ERROR);
 			c.setError(this.getException());
+			tasksToExecute.remove(this);
 		}
```

Comment:
use `protected void done()` for removing the task?

Reply:
I was unsure if the task fails the method is also executed. A small test showed that it is always executed, so i'll use it.

Reply:
Fixed in 419eb93